src/components/TaskModal/TaskModal.tsx

```diff
@@ -0,0 +1,220 @@
+import { Box, Button, Grid, MenuItem, Modal, TextField } from '@mui/material';
+import { useFormik } from 'formik';
+import { Task, TaskPriority } from '@interfaces/task';
+import { TEST_USER } from '@interfaces/user';
+
+export type ModalMode = 'Create' | 'Edit';
+
+interface TaskModal {
+    modalMode: ModalMode;
+    isModalOpen: boolean;
```

Comment:
it's just convention but this is typically something like show or shouldShowModal, this is a modal component so you don't need the word modal in lines 10/11